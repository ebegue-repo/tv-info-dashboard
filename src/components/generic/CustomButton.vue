<template>
    <button 
        type="submit" 
        :class="[positiveButton ? 'bg-tvm-green' : 'bg-stone-900',
                 'text-white',
                 'inline-flex',
                 'justify-center',
                 'rounded-md',
                 'py-2',
                 'px-5',
                 'mr-3',
                 'text-sm',
                 'font-medium',
                 'shadow-sm', 
                 'focus:outline-none',
                 'focus:ring-2',
                 'focus:ring-indigo-500',
                 'focus:ring-offset-2']">
        {{title}}
    </button>
</template>

<script>
    export default {
      name: 'CustomButton',
      props: {
        title: {
          type: String,
          required: true,
        },
        positiveButton: {
          type: Boolean,
          required: false
        }
      }
    }
</script>