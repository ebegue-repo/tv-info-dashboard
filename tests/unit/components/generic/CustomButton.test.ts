import { mount } from '@vue/test-utils';
import CustomButton from '@/components/generic/CustomButton.vue';

describe('CustomButton', () => {
    test('It should render a button with title "test label"', () => {
        const wrapper = mount(CustomButton, {
            props: {
                title: "test label",
                positiveButton: false
            }
        });
        
        expect(wrapper.text()).toBe("test label");
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("It should use the bg-tvm-green class and match the snapshot", () => {
        const wrapper = mount(CustomButton, {
          props: {
            title: "test label",
            positiveButton: true
          },
        });
        const button = wrapper.find("button");
        
        expect(button.classes("bg-tvm-green")).toBe(true);
        expect(button.classes("text-white")).toBe(true);
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("It should use the bg-stone-900 class and match the snaphshot", () => {
        const wrapper = mount(CustomButton, {
          props: {
            title: "test label",
            positiveButton: false
          },
        });
        const button = wrapper.find("button");
        expect(button.classes("bg-stone-900")).toBe(true);
        expect(button.classes("text-white")).toBe(true);
        expect(wrapper.html()).toMatchSnapshot();
    });
});