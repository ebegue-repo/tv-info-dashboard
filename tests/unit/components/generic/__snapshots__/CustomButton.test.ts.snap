// Vitest Snapshot v1

exports[`CustomButton > It should render a button with title "test label" 1`] = `"<button type=\\"submit\\" class=\\"bg-stone-900 text-white inline-flex justify-center rounded-md py-2 px-5 mr-3 text-sm font-medium shadow-sm focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2\\">test label</button>"`;

exports[`CustomButton > It should use the bg-stone-900 class and match the snaphshot 1`] = `"<button type=\\"submit\\" class=\\"bg-stone-900 text-white inline-flex justify-center rounded-md py-2 px-5 mr-3 text-sm font-medium shadow-sm focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2\\">test label</button>"`;

exports[`CustomButton > It should use the bg-tvm-green class and match the snapshot 1`] = `"<button type=\\"submit\\" class=\\"bg-tvm-green text-white inline-flex justify-center rounded-md py-2 px-5 mr-3 text-sm font-medium shadow-sm focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2\\">test label</button>"`;
